bus/src/main/java/org/hawkular/metrics/component/publish/InsertedDataSubscriber.java

```diff
@@ -131,5 +149,43 @@ void shutdown() {
         if (subscription != null) {
             subscription.unsubscribe();
         }
+        publishTask.cancel();
+        publishTimer.cancel();
+    }
+
+    private int getPublishPeriod() {
+        try {
+            return Integer.parseInt(publishPeriodProperty);
+        } catch (NumberFormatException e) {
+            log.warnf("Invalid publish period. Setting default value %s", METRICS_PUBLISH_PERIOD.defaultValue());
+            return Integer.parseInt(METRICS_PUBLISH_PERIOD.defaultValue());
+        }
+    }
+
+    private class PublishTask extends TimerTask {
```

Comment:
https://gist.github.com/burmanm/040c95f71c71dbc36f3b8a9a5367aa53

That should accomplish everything you've done to this class. If you want the buffer size to be configurable, then you'll need to return those settings, although I'm not sure why you want the buffer window at all (does it help with JMS performance issues?)